li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -534,27 +694,17 @@ public void uncaughtException(Thread t, Throwable e) {
     try {
       if (!countDownLatch.await(deadlineTimeMs - System.currentTimeMillis(), TimeUnit.MILLISECONDS)) {
         LiKafkaClientsUtils.dumpStacksForAllLiveThreads(threads);
-        throw new KafkaException("Fail to close all consumers for cluster group " + _clusterGroup + " in " +
-            timeout + " " + timeUnit);
+        throw new KafkaException("fail to perform " + methodName + " for cluster group " + _clusterGroup + " in " +
```

Comment:
for bonus points could iterate over threads still alive and print their stacks

Reply:
I think that's what the line above (LiKafkaClientsUtils.dumpStacksForAllLiveThreads(threads)) is doing? Did you mean something else?